Classes/Application/ReloadNodes/ReloadNodesQueryHandler.php

```diff
@@ -0,0 +1,172 @@
+            throw new NoSiteNodeWasFound(
+                sprintf(
+                    'The site node with NodeAddress "%s" was not found.',
+                    $query->siteId->value
+                ),
+                1712584589
+            );
+        }
+
+        $ancestors = $subgraph->findAncestorNodes(
+            $documentNode->nodeAggregateId,
+            FindAncestorNodesFilter::create(
+                NodeTypeCriteria::fromFilterString(NodeTypeNameFactory::NAME_DOCUMENT)
+            )
+        );
+
+        $nodeMapBuilder = NodeMap::builder(
+            MinimalNodeForTree::class,
+            $this->nodeInfoHelper,
+            $actionRequest
+        );
+        $nodeMapBuilder->addNode($siteNode);
+
+        $gatherNodesRecursively = function (
+            &$nodeMapBuilder,
+            Node $baseNode,
+            $level = 0
+        ) use (
+            &$gatherNodesRecursively,
+            $baseNodeTypeConstraints,
+            $query,
+            $ancestors,
+            $subgraph
+        ) {
+            if ($level < $this->loadingDepth || // load all nodes within loadingDepth
+                $this->loadingDepth === 0 || // unlimited loadingDepth
+                // load toggled nodes
+                $query->toggledNodesIds->contain($baseNode->nodeAggregateId) ||
+                // load children of all parents of documentNode
+                in_array($baseNode->nodeAggregateId->value, array_map(
+                    fn (Node $node): string => $node->nodeAggregateId->value,
+                    iterator_to_array($ancestors)
+                ))
+            ) {
+                foreach ($subgraph->findChildNodes(
+                    $baseNode->nodeAggregateId,
+                    FindChildNodesFilter::create(nodeTypes: $baseNodeTypeConstraints)
+                ) as $childNode) {
+                    $nodeMapBuilder->addNode($childNode);
+                    $gatherNodesRecursively($nodeMapBuilder, $childNode, $level + 1);
+                }
+            }
+        };
+        $gatherNodesRecursively($nodeMapBuilder, $siteNode);
+
+        $nodeMapBuilder->addNode($documentNode);
+
+        foreach ($query->clipboardNodesIds as $clipboardNodeId) {
+            // TODO: does not work across multiple CRs yet.
+            $clipboardNode = $subgraph->findNodeById($clipboardNodeId);
+            if ($clipboardNode) {
+                $nodeMapBuilder->addNode($clipboardNode);
+            }
+        }
+
+        /* TODO: we might use the Subtree as this may be more efficient
+         - but the logic above mirrors the old behavior better.
+        if ($loadingDepth === 0) {
+            throw new \RuntimeException('TODO: Loading Depth 0 not supported');
+        }
+        $subtree = $contentSubgraph->findSubtree([$siteNode], $loadingDepth, $nodeTypeConstraints);
+        $subtree = $subtree->getChildren()[0];
+        $this->flattenSubtreeToNodeList($nodeAccessor, $subtree, $nodeMapBuilder);*/
```

Comment:
```suggestion
        /* TODO: we might use the Subtree as this may be more efficient
         - but the logic above mirrors the old behavior better.
         https://github.com/neos/neos-ui/issues/3517#issuecomment-2070274053 */
```